manifests/vrrp/instance.pp

```diff
@@ -167,45 +173,81 @@
-  $notify_script                                                    = undef,
-  $multicast_source_ip                                              = undef,
-  $unicast_source_ip                                                = undef,
-  $unicast_peers                                                    = undef,
-  $dont_track_primary                                               = false,
-  $use_vmac                                                         = false,
-  $vmac_xmit_base                                                   = true,
-  Boolean $native_ipv6                                              = false,
-
+  $virtual_ipaddress                                                      = undef,
+  $auth_type                                                              = undef,
+  $auth_pass                                                              = undef,
+  $track_script                                                           = undef,
+  Optional[Array[String[1]]] $track_process                               = undef,
+  $track_interface                                                        = undef,
+  $lvs_interface                                                          = undef,
+  $virtual_ipaddress_int                                                  = undef,
+  $virtual_ipaddress_excluded                                             = undef,
+  $virtual_routes                                                         = undef,
+  Optional[Array[Keepalived::Vrrp::Instance::VRule]] $virtual_rules       = undef,
+  $smtp_alert                                                             = false,
+  $nopreempt                                                              = false,
+  $preempt_delay                                                          = undef,
+  $advert_int                                                             = 1,
+  $garp_master_delay                                                      = 5,
+  $garp_master_refresh                                                    = undef,
+  Optional[Integer] $garp_lower_prio_repeat                               = undef,
+  Optional[Boolean] $higher_prio_send_advert                              = undef,
+  Optional[Stdlib::Absolutepath] $notify_script_master_rx_lower_pri       = undef,
+  $notify_script_master                                                   = undef,
+  $notify_script_backup                                                   = undef,
+  $notify_script_fault                                                    = undef,
+  $notify_script_stop                                                     = undef,
+  $notify_script                                                          = undef,
+  $multicast_source_ip                                                    = undef,
+  Optional[Stdlib::IP::Address] $unicast_source_ip                        = undef,
+  Variant[Array[Stdlib::IP::Address], Stdlib::IP::Address] $unicast_peers = [],
```

Comment:
I think you want to keep `Optional` as the default had `undef` and this changes that behavior.

Reply:
This was intentional, to change the default from `undef` to `[]`.